hapi/error.py

```diff
@@ -73,29 +78,32 @@ def __unicode__(self):
 
     def _dict_vals_to_unicode(self, data):
         unicode_data = {}
-        for key, val in data.items():
-            if not isinstance(val, basestring):
-                unicode_data[key] = unicode(val)
-            elif not isinstance(val, unicode):
-                unicode_data[key] = unicode(val, 'utf8', 'ignore')
-            else:
+        for key, val in list(data.items()):
```

Comment:
I don't think that this new code does the same as the old code for python2. In particular, if passed a `str`, it will put a `str` into the `unicode_data` dict rather than a `unicode`

Reply:
I imported ```str``` from ```builtins``` at the top of the file. So now in python 2, ```str``` will also mean ```unicode```, then it will go to the second case ```unicode_data[key] = val.decode('utf-8', 'ignore')```. Maybe I should name it as ```unicode``` to avoid confusion? I just thought ```unicode``` is no longer available in python3, and ```str``` would be the final name for the previous ```unicode```.

Reply:
I think that renaming str to mean unicode is very dangerous -- it certainly confused me.

Reply:
Already retained the original ```unicode()``` function by this line ```from builtins import str as unicode```.